src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -848,11 +862,12 @@ ReactDOMComponent.Mixin = {
         CONTENT_TYPES[typeof props.children] ? props.children : null;
       var childrenToUse = contentToUse != null ? null : props.children;
       if (contentToUse != null) {
-        // TODO: Validate that text is allowed as a child of this node
-        if (__DEV__) {
-          setAndValidateContentChildDev.call(this, contentToUse);
+        if (shouldSetNodeTextContent(lazyTree, props, contentToUse)) {
```

Comment:
Sounds good, I wasn't totally sure if there were situations where setting `textContent` to an empty string was valid.